library/Reporting/Report.php

```diff
@@ -30,98 +31,65 @@ class Report
     /** @var Reportlet[] */
     protected $reportlets;
 
-    /** @var Schedule */
-    protected $schedule;
-
     /** @var Template */
     protected $template;
 
+    /** @var Schedule */
+    protected $schedule;
+
     /**
-     * @param int $id
-     *
-     * @return  static
+     * @param Model\Report $model
      *
-     * @throws  Exception
+     * @return Report
      */
-    public static function fromDb($id)
+    public static function fromModel(Model\Report $model): Report
     {
         $report = new static();
 
-        $db = $report->getDb();
+        $rs = $model::on($report->getDb())
+            ->with([
+                'timeframe',
+                'template',
+                'reportlet',
+                'reportlet.config',
+                'schedule'
+            ])
+            ->filter(Filter::equal('id', $model->id));
 
-        $select = (new Sql\Select())
-            ->from('report')
-            ->columns('*')
-            ->where(['id = ?' => $id]);
+        /** @var $result Model\Report */
+        $result = $rs->first();
```

Comment:
What is this for? You have already created your reporting model within the controller.